File: 1780. Check if Number is a Sum of Powers of Three.java
class Solution {
    int[] powersOfThree = new int[] { 1, 3, 9, 27, 81, 243, 729, 2187, 6561, 19683, 59049,
    177147,531441,1594323,4782969,14348907,43046721,129140163,387420489,1162261467};
    public boolean checkPowersOfThree(int n) {
        if (0 == n) {
            return false;
        }
        for (int i = powersOfThree.length-1; i >= 0; i--) {
            if (n >= powersOfThree[i]) {
                n -= powersOfThree[i];
            }
        }
        return 0 == n;
    }
}

File: 1780. Check if Number is a Sum of Powers of Three.rs
impl Solution {
    pub fn check_powers_of_three(n: i32) -> bool {
        let mut max_power = 4782969;
        let mut n = n;
        if n == 0 { return false; }
        while (max_power > 0 && n > 0) {
            if (n >= max_power) {
                n -= max_power;
            }
            max_power /= 3;
        }
        n == 0
    }
}
